cukd/data.h

```diff
@@ -33,27 +33,27 @@ namespace cukd {
 
   
   /*! defines an abstract interface to what a 'node' in a k-d tree
-    is. This needs to define the follwing:
+    is. This needs to define the following:
 
     - data_traits::scalar_t: the scalar type of each point member (eg,
     float for a float3 data_t)
 
+    NOTE THIS IS NOT USED ANYWHERE
```

Comment:
everywhere you are using num_dims_of instead
which means that one would have to specialize num_dims_of for a custom type. IMO it would be better to have everything in the point trait type, and even get rid of those free floating trait functions in math.h (scalar_type_of, num_dims_of, get_coord, etc...)

Reply:
agreed; that's actually what i (had to) do in the iw/host_builder branch becasue the specialization of these num_dims and scalar_t turned out to be rather tricky.
i'll merge new branch later today - that new branch does things a bit differently: there's a cukd::point_traits<> template that lets define how a given class can be used for point-related math; and then a separate data_traits that gets passed to the builder and traversers to define how a given data record represents its data.